core/db/db.go

```diff
@@ -172,32 +146,23 @@ func getDBPath() string {
 	return dbPath + "db"
 }
 
-func createDBIfDBPathEmpty() error {
-	dbPath := getDBPath()
-	missing, err := dirMissingOrEmpty(dbPath)
-	if err != nil {
-		return err
-	}
-	dbLogger.Debug("Is db path [%s] empty [%t]", dbPath, missing)
-	if missing {
-		err := CreateDB()
-		if err != nil {
-			return nil
-		}
+// Open open underlying rocksdb
+func (openchainDB *OpenchainDB) Open() {
+	openchainDB.mux.Lock()
+	if openchainDB.dbState == opened {
+		openchainDB.mux.Unlock()
+		return
 	}
-	return nil
-}
 
-func openDB() (*OpenchainDB, error) {
-	if isOpen {
-		return openchainDB, nil
-	}
+	defer openchainDB.mux.Unlock()
 
 	dbPath := getDBPath()
+	mkDirIfNotExist(dbPath)
+
 	opts := gorocksdb.NewDefaultOptions()
 	defer opts.Destroy()
 
-	opts.SetCreateIfMissing(false)
+	opts.SetCreateIfMissing(true)
```

Comment:
Yes, panic is fine instead of throwing error in DB creation. However, there were some discussions of revising the errors and their handling so I am not sure whether we should take things one by one or do we want a holistic approach. The main doubt in mind is what are the different handling we want is it only a choice between throw error and panic?

The reason of closing the db in the create method was to handle the columnfamily related stuff at one place i.e., `openDB` method.